src/app/qgssnappingwidget.cpp

```diff
@@ -287,6 +287,14 @@ QgsSnappingWidget::QgsSnappingWidget( QgsProject *project, QgsMapCanvas *canvas,
   tracingMenu->addAction( widgetAction );
   mEnableTracingAction->setMenu( tracingMenu );
 
+  // self-snapping button
+  mSelfSnappingAction = new QAction( tr( "Self-snapping" ), this );
+  mSelfSnappingAction->setCheckable( true );
+  mSelfSnappingAction->setIcon( QIcon( QgsApplication::getThemeIcon( "/mIconSnappingSelf.svg" ) ) );
+  mSelfSnappingAction->setToolTip( tr( "Enable Self-snapping" ) );
```

Comment:
```suggestion
  mSelfSnappingAction->setToolTip( tr( "If self snapping is enabled, snapping will also use the current state of the digitized feature into consideration." ) );
```

I think the tool tip can be a bit more verbose.